File: backend/app/models/paper.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

PaperSource = Literal["arxiv", "pubmed", "upload"]


class Paper(BaseModel):
    id: str
    title: str
    authors: list[str] = Field(default_factory=list)
    abstract: str
    published: datetime
    updated: datetime
    pdf_url: str
    source: PaperSource = "arxiv"


class SearchResponse(BaseModel):
    query: str
    count: int
    papers: list[Paper]

File: frontend/types/paper.ts
export type PaperSource = "arxiv" | "pubmed" | "upload";

export interface Paper {
  id: string;
  title: string;
  authors: string[];
  abstract: string;
  published: string;
  updated: string;
  pdf_url: string;
  source: PaperSource;
}

export interface SearchResponse {
  query: string;
  count: number;
  papers: Paper[];
}
